bigbang/archive.py

```diff
@@ -340,3 +357,179 @@ def clean_footer(foot):
     )
 
     return candidates[0:number]
+
+
```

Comment:
Given that we are refactoring, maybe this is a good time to clear up the confusion of these two functions. 

Reading over this code again, it seems like `load_data` and `open_list_archives` actually perform the same basic task, just that `load_data` only opens CSV files and `open_list_archives` looks for and opens MBOX files with various filenames. The `mbox` parameter in both doesn't seem to do much (for load_data it switches to `open_list_archives` and for open_list_archives it's just a flag for interpreting the path parameter as a single file name).

Could we have a single function that loads the already-collected mailing list archives for a particular list and returns it as a dataframe? And then internally bigbang can detect whether it's stored as a csv or a bunch of mbox files?

Reply:
This sounds like a good idea to me. But I think maybe it should be tied to a more thoroughly designed way of routing data storage.

See #501. I think we would like to be able to refer to mailing lists by a shorthand name, but then we need their full identifiers, like a URL, for data collection in some cases.

I can imagine a case where somebody pulls data from multiple sources -- IETF, 3GPP, W3C -- and there is a name conflict for the "short name" of the list.

That suggests that every method of data ingress should be populating a subdirectory of the archive directory. (One day, maybe these methods should be populating an actual database rather than dumping the flat files to disk).

This code was all written when BigBang was mainly about scraping Mailman instances. So it's all very ripe to be rewritten and redesigned for the newer use cases.

Reply:
#512